src/com/goide/refactor/GoIntroduceVariableBase.java

```diff
@@ -0,0 +1,252 @@
+      if (callReference != null) {
+        String name = StringUtil.decapitalize(callReference.getIdentifier().getText());
+        if (name.startsWith("get")) {
+          name = name.substring(3);
+        }
+        else if (name.startsWith("is")) {
+          name = name.substring(2);
+        }
+        for (int i = 0; i < name.length(); i++) {
+          if (i == 0 || (Character.isLowerCase(name.charAt(i - 1)) && Character.isUpperCase(name.charAt(i)))) {
+            names.add(StringUtil.decapitalize(name.substring(i)));
+          }
+        }
+      }
+      return ContainerUtil.newLinkedHashSet(ContainerUtil.reverse(names));
+    }
+    return ContainerUtil.newLinkedHashSet("i");
+  }
+
+  protected static void showCannotPerform(@NotNull Project project, @NotNull Editor editor) {
+    showCannotPerform(project, editor, null, null);
+  }
+
+  protected static void showCannotPerform(@NotNull Project project, @NotNull Editor editor, String message, String title) {
+    message = StringUtil.isEmpty(message) ? "Cannot Perform Refactoring" : message;
+    title = StringUtil.isEmpty(title) ? "Cannot Perform Refactoring" : title;
+    CommonRefactoringUtil.showErrorHint(project, editor, message, title, "refactoring.extractVariable");
+  }
+
+  protected static void smartIntroduce(@NotNull final Project project, @NotNull final Editor editor, @NotNull List<GoExpression> expressions) {
+    IntroduceTargetChooser.showChooser(editor, expressions, new Pass<GoExpression>() {
+      @Override
+      public void pass(GoExpression expression) {
+        performOnElement(project, editor, expression);
+      }
+    }, new Function<GoExpression, String>() {
+      @Override
+      public String fun(GoExpression expression) {
+        return expression.getText();
+      }
+    });
+  }
+
+  private static GoExpression findParentExpression(@NotNull PsiElement element) {
+    return PsiTreeUtil.getParentOfType(element, GoExpression.class);
+  }
+
+  private static List<GoExpression> collectNestedExpressions(@NotNull PsiElement element) {
+    GoExpression expression = element instanceof GoExpression ? (GoExpression)element : findParentExpression(element);
+    List<GoExpression> expressions = ContainerUtil.newArrayList();
+    while (expression != null) {
+      if (!(expression instanceof GoParenthesesExpr) && GoInspectionUtil.getExpressionResultCount(expression) == 1) {
+        expressions.add(expression);
+      }
+      expression = findParentExpression(expression);
+    }
+    return expressions;
+  }
+
+  protected static List<GoExpression> collectExpressionsAtOffset(@NotNull final PsiFile file,
+                                                                 @NotNull final Document document,
+                                                                 int offset) {
+    CharSequence text = document.getCharsSequence();
+    if (offset >= text.length()) return ContainerUtil.emptyList();
+    if (text.charAt(offset) == ')') offset--;
```

Comment:
I ignore GoParenthesesExpr in collectNestedExpressions method so I don't want the initial expression to be ignored when there is a right parenthesis under the caret.